### PR TITLE
clkinfo: save clock info on `kill`/`remove`

### PR DESCRIPTION
**Pull request by bobrippling**
A follow-up to #3234, save clkinfo settings so on a fresh load we display what was last shown

### CODE REVIEW

**Comment by bobrippling** [comment]
@gfwilliams interested in your thoughts on this, or whether we detect this after a `focus`/`blur` (or even `show`/`hide`/`run`) and do it automatically. I think this is better since it's unlikely most clkinfos will want this functionality

**Comment by gfwilliams** [comment]
Yeah... So right now we save whenever the user swipes?

I think this change is probably ok - we don't want to save whenever we change as I guess we may end up doing a bunch more writes than we need.

**Or...** we just save the state in `E.on('kill'` and `.remove`, and nowhere else? That is definitely preferable - no messing around writing all the time, and we're sure the state will be saved and then none of the clockinfos have to worry about it?

**Comment by bobrippling** [comment]
Yeah exactly, we only save after a swipe currently. I prefer the idea of saving on `kill` and `remove` in fact, I'll alter this and that avoids apps/widgets having to manage it too

File: apps/clock_info/lib.js [hunk 1]
```diff
@@ -333,6 +336,7 @@ exports.addInteractive = function(menu, options) {
   menuShowItem(menu[options.menuA].items[options.menuB]);
   // return an object with info that can be used to remove the info
   options.remove = function() {
+    save();
```
Comment (gfwilliams) [inline apps/clock_info/lib.js:339]: we'd need to do `E.removeListener("kill",save)` here too I reckon or we will have a memory leak when doing fast load

**Comment by gfwilliams** [comment]
Looks good apart from that possible memory leak? Might be worth running a fast load clock with it and just checking nothing gets leaked even after that change though

File: apps/clock_info/lib.js (hunk 1 above)
Comment (bobrippling) [reply apps/clock_info/lib.js:339]: Good catch, thanks!

**Comment by bobrippling** [comment]
Thanks - shall run it on a fastload clock tomorrow and confirm the memory load

**Comment by bobrippling** [comment]
It seems stable - this was with a clockface with two clock infos:

```javascript
>process.memory(true)
={ free: 6756, usage: 5244,  ... }

>load()
>process.memory(true)
={ free: 6756, usage: 5244,  ... }

>load()
>process.memory(true)
={ free: 6756, usage: 5244,  ... }

>load()
>process.memory(true)
={ free: 6756, usage: 5244,  ... }

>load()
>process.memory(true)
={ free: 6756, usage: 5244,  ... }

>load()
>process.memory(true)
={ free: 6756, usage: 5244,  ... }
```

**Comment by gfwilliams** [comment]
Looks great, thanks!

**Pull request merged**